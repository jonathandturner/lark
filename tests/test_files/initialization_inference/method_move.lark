//~ execute:no

// Test that we can move from `bar.foo` and then reinitialize `bar`
// and the code still compiles.

struct Foo {
    x: uint

    // FIXME: Eventually, this should probably be `own take()` or something,
    // because it will take ownership of the self argument. But right now
    // we make all arguments be "own" mode.
    take() {
    }
}

def main() {
  let foo = Foo(x: 22)
  foo.take()
  foo.take()
  //~ ERROR
}
