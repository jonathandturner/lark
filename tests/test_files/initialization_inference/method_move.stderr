error: access to uninitialized path
- initialization_inference/method_move:19:2
19 |   foo.take()
   |   ^^^
